bentoml/_internal/io_descriptors/numpy.py

```diff
@@ -237,6 +237,67 @@ async def to_http_response(self, obj: ext.NpNDArray, ctx: Context | None = None)
         else:
             return Response(json.dumps(obj.tolist()), media_type=MIME_TYPE_JSON)
 
+    def handle_tuple(self, proto_tuple):
+        """
+        Convert given protobuf tuple to a tuple list
+        """
+        import io_descriptors_pb2
+        from google.protobuf.timestamp_pb2 import Timestamp
+        from google.protobuf.duration_pb2 import Duration
+
+        tuple_arr = []
+        [tuple_arr.append(i) for i in getattr(proto_tuple, "value_")]
+
+        if(not tuple_arr):
+            raise ValueError("Provided tuple is either empty or invalid.")
+
+        return_arr = []
+
+        for i in range(len(tuple_arr)):
+            val = getattr(tuple_arr[i], tuple_arr[i].WhichOneof("dtype"))
+
+            if(tuple_arr[i].WhichOneof("dtype") == "timestamp_"):
+                val = Timestamp.ToDatetime(val)
+            elif(tuple_arr[i].WhichOneof("dtype") == "duration_"):
+                val = Duration.ToTimedelta(val)
+
+            if(type(val) == io_descriptors_pb2.NumpyNdarray):
+                val = self.proto_to_arr(val)
+            elif(type(val) == io_descriptors_pb2.Tuple):
+                val = self.handle_tuple(val)
```

Comment:
you should also handle exception here. What happens if the val doesn't belong to the previous two types?
```python
                val = self.handle_tuple(val)
            else:
                ...
```

Reply:
The `else` case would not apply here because if it is not a tuple or an array, it is one of the other types given `val` is not empty or invalid. I added an exception check in my latest commit for `val` to verify that it is valid before doing the type checks.

Reply:
noted.